src/vsphere_cpi/spec/unit/cloud/vsphere/cloud_spec.rb

```diff
@@ -785,6 +788,267 @@ module VSphereCloud
             environment
           )
         end
+
+        context 'when VM has duplicate security groups specified in vm_type, lbs and BOSH Groups' do
+          let(:vm_type) do
+            {
+              'cpu' => 1,
+              'ram' => 1024,
+              'disk' => 4096,
+              'nsx' => {
+                'security_groups' => [fake_duplicate_sg, another_fake_duplicate_sg, another_fake_duplicate_sg],
+                'lbs' => [
+                  {
+                    'edge_name' => 'fake-edge',
+                    'pool_name' => 'fake-pool',
+                    'security_group' => fake_duplicate_sg,
+                    'port' => 443,
+                    'monitor_port' => 443,
+                  },
+                  {
+                    'edge_name' => 'fake-edge-2',
+                    'pool_name' => 'fake-pool-2',
+                    'security_group' => fake_duplicate_sg,
+                    'port' => 443,
+                    'monitor_port' => 443,
+                  }
+                ]
+              }
+            }
+          end
+          let(:environment) do
+            {
+              'bosh' => {
+                'groups' => [
+                  fake_duplicate_sg,
+                  another_fake_duplicate_sg,
+                  fake_unique_bosh_group,
+                ]
+              }
+            }
+          end
+          it 'should de-duplicate the security tags and attach them to the VM' do
+            allow(VmConfig).to receive(:new).and_return(vm_config)
+            allow(NsxHttpClient).to receive(:new)
+              .with('fake-nsx-user', 'fake-nsx-password', 'fake-log-file')
+              .and_return(http_basic_auth_client)
+            allow(NSX).to receive(:new).and_return(nsx)
+
+            expect(VmCreator).to receive(:new).and_return(vm_creator)
+            expect(vm_creator).to receive(:create).with(vm_config).and_return(fake_vm)
+            expect(cloud_config).to receive(:validate_nsx_options)
+            expect(nsx).to receive(:add_vm_to_security_group).exactly(3).times
```

Comment:
you could also add a check expect(nsx).to receive(:add_vm_to_security_group).with(/(fake_duplicate_sg| another_fake_duplicate_sg|fake_unique_bosh_group)/, fake_vm.mob_id)

Reply:
Yeah. that would be nice. I will add it to this commit but not for the already back ported once.